### ISSUE TITLE
IfcConvert & Segmentation Fault in 0.7.0 (with multithreading)

### ISSUE THREAD

**Issue by bdaniere** [IfcOpenShell/IfcOpenShell#1895]
Hello,

I am upgrading IfcOpenShell from version 0.6.0 to version 0.7.0 on an ubuntu environment 
I regularly encounter Segmentation Fault when converting IFC --> GLB while these files did not present any problem on version 0.6.0
_IfcOpenShell IfcConvert v0.7.0-04535d27 (OCC 7.5.3)_

For example, the Marc Antoine Petit.ifc model ([main-buildings-a17-dp-49-20170119-014534.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/7596949/main-buildings-a17-dp-49-20170119-014534.zip)) crashes when creating geometries

Here is the command I use for the conversion: 
`IfcConvert /home/benjamin/MAP.ifc /home/benjamin/MAP.glb --use-element-guids --no-progress -y --enable-layerset-slicing --threads 12 --include entities IfcObject --exclude entities IfcOpeningElement`

It seems that the conversion goes smoothly if I remove the **--thread** 
(I have also tried to replace the value 12 with 2, but the problem persists)

To detect if there is any problem that could explain this error, I also did a type by type conversion
for the walls, for example :

`IfcConvert /home/benjamin/toto.ifc /home/benjamin/toto.glb --use-element-guids --no-progress -y --enable-layerset-slicing --threads 12 --include entities IfcWallStandardCase`

The following Ifc types generate an error: 
```
conversion failed with IfcWall
conversion failed with IfcStairFlight
conversion failed with IfcWallStandardCase
```
I also tried to perform an instance by instance conversion to identify the instances that could generate a potential error 
I have no problem with this approach (except that it is a bit more time consuming):
`
IfcConvert /home/benjamin/toto.ifc /tmp/140215118833216.glb --use-element-guids --no-progress -y --enable-layerset-slicing --include attribute GlobalId 1T2hYTZj1EsBZaf4hQmbKK`

**Comment by aothms** [comment]
Yes I can confirm that an empty result triggered the notification that an element was ready. But I didn't have the time to investigate why the element result was empty. Perhaps you can double check if you see missing elements after the fix, comparing 0.6 and 0.7?
